### PR TITLE
PLNSRVCE-255 - sharedSecret for infra-deployments update

### PR DESCRIPTION
**Pull request by mbharatk**
Instead of creating a secret in each service namespace, a shared secret is defined. (https://github.com/redhat-appstudio/infra-deployments/pull/350)
this PR adds the shared secret config to update-infra-deployments task, which can be used to update infra-deployments.

### CODE REVIEW

**Comment by gabemontero** [comment]
this should not merge until we merge https://github.com/redhat-appstudio/infra-deployments/pull/349 minimally

we quite possibly will want also gate this on our sorting out of the 3 minute delay wrt permission application for the pipeline SA that 
@sbose78 @Michkov and I have been discussing in slack

let's see if 

/hold

is set up for this repo

**Comment by gabemontero** [comment]
> 
> let's see if
> 
> /hold
> 
> is set up for this repo

does not look like it :-)

**Comment by mbharatk** [comment]
@gabemontero Sure, we could keep this on hold until the #349 is merged.

File: tasks/update-infra-deployments.yaml
Comment (Michkov) [inline tasks/update-infra-deployments.yaml:33]: do not use /tmp/ that could be handled differently then other dirs. Rather use `/secrets/deploy-key`

File: tasks/update-infra-deployments.yaml
Comment (Michkov) [inline tasks/update-infra-deployments.yaml:59]: should be path directly (not evaluated as `$()`) -> `value: /secrets/deploy-key/appstudio-staging-ci`

File: tasks/update-infra-deployments.yaml
Comment (Michkov) [inline tasks/update-infra-deployments.yaml:35]: `${WORKSPACE_SSH_DIRECTORY_PATH}` has been removed, so it should be replaced by `/secrets/deploy-key`

File: tasks/update-infra-deployments.yaml
Comment (Michkov) [inline tasks/update-infra-deployments.yaml:57]: the volume is not mounted in this step

**Review by Michkov — CHANGES_REQUESTED** [review]

File: tasks/update-infra-deployments.yaml
Comment (Michkov) [inline tasks/update-infra-deployments.yaml:15]: Also please add comment that the workspace is only as placeholder and can be removed after it will be removed from consumers.

File: tasks/update-infra-deployments.yaml
Comment (Michkov) [inline tasks/update-infra-deployments.yaml:20]: Put this description as comment to the volume

**Review by Michkov — CHANGES_REQUESTED** [review]

**Review by Michkov — APPROVED** [review]

**Pull request merged**
